src/main/java/racingcar/control/input/StandardInput.java

```diff
@@ -1,17 +1,36 @@
 package racingcar.control.input;
 
+import racingcar.control.input.validator.NumberValidator;
 import racingcar.view.View;
 
 import java.util.InputMismatchException;
+import java.util.Objects;
 import java.util.Scanner;
 
 public abstract class StandardInput {
     private static final Scanner scanner = new Scanner(System.in);
+    private NumberValidator validator;
+
+    public StandardInput() {
+    }
+
+    public StandardInput(NumberValidator validator) {
+        this.validator = validator;
+    }
 
     abstract protected View view();
 
     public int getInt() {
         this.view().render();
+
+        int result = input();
+        if (Objects.nonNull(validator)) {
+            validator.verify(result);
+        }
```

Comment:
```suggestion
            validator.verify(result);
```
if 문을 제거하고 더미 validator를 두는것도 좋아보여요
```
public class DummyValidator implements NumberValidator {
    @Override
    public <T extends Number> void verify(T value) {
        // do nothing
    }
}
```
```
public StandardInput() {
    this.validator = new DummyValidator();
}
```